pkg/moov/bank_account.go

```diff
@@ -57,15 +59,112 @@ type ACHStatusUpdates struct {
+	Account   *BankAccount `json:"account,omitempty"`
+	Plaid     *Plaid       `json:"plaid,omitempty"`
+	PlaidLink *PlaidLink   `json:"plaidLink,omitempty"`
+	MX        *MX          `json:"mx,omitempty"`
+}
+
+// Plaid is a direct to plaid integration. Use this if you signed an agreement with Plaid directly.
+type Plaid struct {
+	// Token is the plaid processor_token
+	Token string `json:"token"`
+}
+
+// PlaidLink is a Moov-managed Plaid integration. Use this if you purchase Plaid through Moov.
+type PlaidLink struct {
+	// PublicToken is the plaid public_token
+	PublicToken string `json:"publicToken"`
+}
+
+// MX is authorization code of a MX account which allows a processor to retrieve a linked payment account.
+type MX struct {
+	AuthorizationCode string `json:"authorizationCode"`
+}
+
+type CreateBankAccountType callArg
+
+func WithBankAccount(bankAccount BankAccount) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		bankAccountJSON, err := json.Marshal(bankAccount)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
+}
+
+func WithPlaidLink(plaidLink PlaidLink) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		// need to use a map to add the top level "plaidLink" json key. expected format is:
+		//	{
+		// 		"plaidLink": {
+		// 			"publicToken": "insert token"
+		// 		}
+		// 	}
+
+		plaidLinkMap := map[string]PlaidLink{
+			"plaidLink": plaidLink,
+		}
+		bankAccountJSON, err := json.Marshal(plaidLinkMap)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
+}
+
+func WithPlaid(plaid Plaid) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		// need to use a map to add the top level "plaid" json key. expected format is:
+		//	{
+		// 		"plaid": {
+		// 			"token": "insert token"
+		// 		}
+		// 	}
+
+		plaidMap := map[string]Plaid{
+			"plaid": plaid,
+		}
+		bankAccountJSON, err := json.Marshal(plaidMap)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
+}
+
+func WithMX(mx MX) CreateBankAccountType {
+	return callBuilderFn((func(call *callBuilder) error {
+		// need to use a map to add the top level "mx" json key. expected format is:
+		//	{
+		// 		"mx": {
+		// 			"authorizationCode": "testing"
+		// 		}
+		// 	}
+
+		mxMap := map[string]MX{
+			"mx": mx,
+		}
+		bankAccountJSON, err := json.Marshal(mxMap)
+		if err != nil {
+			return err
+		}
+		call.body = bytes.NewReader(bankAccountJSON)
+		return nil
+	}))
 }
 
 // CreateBankAccount creates a new bank account for the given customer account
-func (c Client) CreateBankAccount(ctx context.Context, accountID string, bankAccount BankAccount) (*BankAccount, error) {
+func (c Client) CreateBankAccount(ctx context.Context, accountID string, opts ...CreateBankAccountType) (*BankAccount, error) {
```

Comment:
should we rename `CreateBankAccount` to `LinkBankAccount` to match what we have there?

we have `POST /accounts/{accountID}/bankaccounts` api reference page as titles "Link a bank account"

https://docs.moov.io/api/sources/bank-accounts/create/